### ISSUE TITLE
[error] Failed to create decryptor: Unable to open URL:

### ISSUE THREAD

**Issue by chovy** [chrippa/livestreamer#1144]
Not sure why this isn't working. I'm passing cookie and proper user agent. It works fine in ffmpeg.

```
  livestreamer \
    -l debug \
    --http-header 'Cookie=${cookie}' \
    --http-header 'User-Agent=iPhone' \
    "hls://${input}" \
    best \
    -o video.ts
```

**Comment by 0880** [comment]
>My suspicion is the headers aren't being passed along to the secondary request for the key.

I don't think so. Headers are passed correctly (also for HLS key requests). I have tested this successfully some time ago.

**Comment by ralyodio** [comment]
would it help if we did a skype session? I'm happy to show you what i'm doing, but due to expiration I can't paste the url/cookies here as it will expired.

**Comment by 0880** [comment]
To quickly check what's going on, you can use tools like [Wireshark] (https://www.wireshark.org). Monitoring your network traffic while Livestreamer is running, will certainly point you in the right direction.

I don't know what's going on in your case, but if the key file inside the playlist points to wrong URL, Livestreamer won't be able to request the key properly (#1097). In this case we would need a command line option to pass a valid URL to the key file.
